Main controller refactoring

mauroz77 opened the pull request.

mauroz77 committed:
Main controller refactoring

- Changes due to the update of the endpoint getmissingdiagnosismappings (from /api/getmissingdiagnosismappings to /api/mappings/getmissingmappings)
- Includes the renaming of url variables in mappingService
- Includes tests for mappingService

mauroz77 committed:
Main controller refactoring

- Fix audit error with instructions from https://itnext.io/fixing-security-vulnerabilities-in-npm-dependencies-in-less-than-3-mins-a53af735261d

mauroz77 committed:
Main controller refactoring

- Use a newer version of init because of error in travis

Afollet commented on src/app/mapping.service.ts:
```diff
@@ -11,13 +11,14 @@ import { SummaryInterface } from './summary-interface';
-    private missingMappingsUrl = this.serverUrl + '/api/getmissingdiagnosismappings';
-    public exportUrl = this.serverUrl + '/api/mappings/export';
-    private uploadURL = this.serverUrl + '/api/mappings/uploads';
+    private SUMMARY_URL = this.BASE_URL + '/summary';
```
What is the reason for the upper-casing? Just a curiosity.

mauroz77 replied:
Just as a convention for constant values. https://google.github.io/styleguide/javascriptguide.xml?showone=Constants#Constants (_Use NAMES_LIKE_THIS for constant values._)

Afollet replied:
Yes, that makes sense. Because of my nativity with Angular I do not understand how these are defined constant.

Afollet replied:
For example, why wouldn't "devServer" be a constant?

mauroz77 replied:
I see your point. That's something that can be improved. We 
could explicitly set them as readonly (I'd say the equivalent to final in Java) https://www.typescriptlang.org/docs/handbook/classes.html#readonly-modifier. I'll see if it works well and update the PR

Afollet commented on sonar-project.properties:
```diff
@@ -1,11 +1,12 @@
 sonar.host.url=http://localhost:9000 
 sonar.login=admin
-sonar.password=admin
+sonar.password=pdx-admin-ui
```
Can you link to information in the Confluence or in the readme how to run SonarQube locally for the project?

mauroz77 replied:
This is the link in confluence: https://www.ebi.ac.uk/seqdb/confluence/display/MouseInformatics/PDX+Admin.
I think putting it in readme is also a good idea but I'll do that in another PR

Afollet reviewed (approved):
The tests look great.

mauroz77 committed:
Main controller refactoring

- Make it explicit that the constants with urls in MappingService are read only
